File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.py
T = int(input())

for x in range(1, T+1):
    n, m = map(int, input().split())
    mat = [list(map(int, input().split())) for i in range(n)]

    result = []

    for i in range(n-m+1):
        for j in range(n-m+1):
            s = 0
            for k in range(m):
                for l in range(m):
                    s += mat[i+k][j+l]
            result.append(s)

    print("#%d %d" %(x, max(result)))

File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.java

import java.util.Scanner;
import java.io.FileInputStream;

public class Solution {
	
	public static void main(String args[]) throws Exception {
		
		Scanner sc = new Scanner(System.in);
		
		int test = sc.nextInt();
		
		for (int t = 1; t <= test; t++) {
			
			int n = sc.nextInt();
			int m = sc.nextInt();
			
			int max = 0;
			
			int[][] mat = new int[n][n];
			
			for (int i= 0; i<n; i++) {
				for (int j = 0; j < n; j++) {
					mat[i][j] = sc.nextInt();
				}
			}
			
			// 인덱스 범위를 넘기는 경
			for (int i = 0; i<n-m+1; i++) {
				for (int j = 0; j <n-m+1; j++) {
					int sum = 0;
					
					for(int k = 0; k <m; k++) {
						for(int l = 0; l <m; l++) {
							sum += mat[i+k][j+l];
						}
					}
					
					if (max < sum) {
						max = sum;
					}
				}
			}
			
			System.out.println("#" + t+ " "+max);
		}
	}

}
